### PR TITLE
Add Travis CI support

### PR DESCRIPTION
**Pull request by hjanos1**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+sudo: required
+
+before_install: sudo apt-get update -qq
+
+install:
+  - sudo apt-get install libboost-filesystem-dev
+  - sudo apt-get install libgtest-dev
+
+before_script:
+  - mkdir gtest_build
```
Comment (dobragab) [inline .travis.yml:13]: Do we have to build gtest ourselves?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - cd gtest_build
+  - cp -r /usr/src/gtest/* ./
+  - cmake CMakeLists.txt
+  - make
+  - sudo cp *.a /usr/lib
+  - cd ..
+  - rm -rf gtest_build
+  - cmake -DWITH_CLING=OFF -DBOOST_FILESYSTEM=ON .
+
+script: make
```
Comment (dobragab) [inline .travis.yml]: Please run tests after building too.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+language: cpp
+
+dist: trusty
```
Comment (dobragab) [inline .travis.yml:3]: Are there more recent distros?

**Review by dobragab — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (hjanos1) [reply .travis.yml:3]: Trusty is the latest Travis CI offers. Beleive me, I tried to get a more recent distro to get better compilers

File: .travis.yml (hunk 1 above)
Comment (hjanos1) [reply .travis.yml:13]: Unfortunately, yes. Ubuntu packages does not come with the required binaries.

File: .travis.yml (hunk 2 above)
Comment (hjanos1) [reply .travis.yml]: See the new commit

**Review by dobragab — APPROVED** [review]

**Pull request merged**